src/nvim/quickfix.c

```diff
@@ -4182,7 +4187,30 @@ static void qf_update_buffer(qf_info_T *qi, qfline_T *old_last)
   qf_update_win_titlevar(qi);
 
   qf_fill_buffer(qf_get_curlist(qi), buf, old_last, qf_winid);
-  buf_inc_changedtick(buf);
+
+  linenr_T new_line_count = buf->b_ml.ml_line_count;
+  colnr_T new_endcol = ml_get_buf_len(buf, new_line_count);
+  bcount_t new_byte_count;
+
+  if (old_last == NULL) {
+    new_byte_count = get_region_bytecount(buf, 1, new_line_count, 0, new_endcol);
+    extmark_splice(buf, 0, 0, old_line_count - 1, 0, old_byte_count, new_line_count - 1,
```

Comment:
is `old_line_count` guaranteed to be >=1 ?

Reply:
initial value of `ml_line_count` is 1 in `ml_open`, It is only set to 0 when buf_freeall is called.